developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,134 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID is needed
+******************
+
+When developing a new audio signal processing component, we
+traditionally implement the component driver with a new unique component
+type declared together, the painpoints with this method including ABI
+alignment among topology/driver/firmware for each new added components,
+and there could be collision if versions are not aligned.
```

Comment:
no, it's a collision between different vendors using the same type for different usages and component.

Reply:
What I wrote in the commit is only about the collision (or I should call it 'unaligned' actually) of different component type, e.g. in some version the type 20 could mean SMART_AMP but in most versions 20 should be DCBLOCK, this doesn't happen frequently, but if the ABI management is not run good enough (that is why @kv2019i has been proposing improvement on it).

For the collision of different vendors using the same type, we talked about this in the discussion meeting also, we have a thought to introduce UUID array for the component driver, which means to list all supported vendors' implementation to the driver (something like the supported PCI IDs in our linux PCI device driver) and use that for the component driver matching. With this, we can still use a single UUID for a specific component type, and no class/subtype mechanism needed. This is still in discussion stage yet, as we don't have real different vendors's implementation for the same driver upstream yet, but we do think of it and leave the possibility of adding support to it.